### ISSUE TITLE
Add option `--remount` for remount support

### ISSUE THREAD

**Issue by Weathercold** [rclone/rclone#6488]
#### What is your current rclone version (output from `rclone version`)?
```console
$ rclone --version
rclone 1.59.2
- os/version: nixos 22.11 (Raccoon) (64 bit)
- os/kernel: 5.19.12-zen1 (x86_64)
- os/type: linux
- os/arch: amd64
- go/version: go1.18.6
- go/linking: dynamic
- go/tags: cmount
```

#### What problem are you are trying to solve?
I'm using rclone as a mount helper. Whenever I try `sudo systemctl reload <rclone-mount>.mount`, it fails because rclone doesn't recognize the flag `--remount`. Normally this can be circumvented by unmounting then remounting, but I also use NixOS and the aforementioned command is automatically invoked whenever I switch system generation.

#### How do you think rclone should be changed to solve that?
Add the `--remount` flag to remount a file system.

#### How to use GitHub
* Please use the 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to show that you are affected by the same issue.
* Please don't comment if you have no relevant information to add. It's just extra noise for everyone subscribed to this issue.
* Subscribe to receive notifications on status change and new comments.

**Comment by ncw** [comment]
> Add the `--remount` flag to remount a file system.

What do you think rclone should when it receives this flag

Here is what I found in the mount man page

> ## remount
>
> Attempt to remount an already-mounted filesystem. This is commonly
> used to change the mount flags for a filesystem, especially to make
> a readonly filesystem writable. It does not change device or mount
> point.
> 
> The remount operation together with the bind flag has special
> semantics. See above, the subsection Bind mounts.
> 
> The remount functionality follows the standard way the mount
> command works with options from fstab. This means that mount does
> not read fstab (or mtab) only when both device and dir are
> specified.
> 
> mount -o remount,rw /dev/foo /dir
> 
> After this call all old mount options are replaced and arbitrary
> stuff from fstab (or mtab) is ignored, except the loop= option
> which is internally generated and maintained by the mount command.
> 
> mount -o remount,rw /dir
> 
> After this call, mount reads fstab and merges these options with
> the options from the command line (-o). If no mountpoint is found
> in fstab, then a remount with unspecified source is allowed.
> 
> mount allows the use of --all to remount all already mounted
> filesystems which match a specified filter (-O and -t). For
> example:
> 
> mount --all -o remount,ro -t vfat
> 
> remounts all already mounted vfat filesystems in read-only mode.
> Each of the filesystems is remounted by mount -o remount,ro /dir
> semantic. This means the mount command reads fstab or mtab and
> merges these options with the options from the command line.

It sounds very hard to diff the config for the mount before and after and work out what to do.

Rclone could
1. ignore the flag
2. attempt to unmount an rclone fs at /mountpoint then carry on as before

?

**Comment by Weathercold** [comment]
I did a few tests using echo as a mount helper:
- `--all`,  loop, and bind mounts are handled by mount itself (not sure if rclone supports loop devices), so no need to support them
- with option `remount`, mount handles merge of options
- with option `remount`, mount does not unmount the mount point automatically, so this is handled by the mount helper
- when attempting to "remount" a device onto a mount point with a different one, it is a no-op and silently returns success

TL;DR rclone only has to verify that the new remote is the same as the old one, then unmount the mount point and carry on as before. If the verification fails then return success (there really should be a warning).

**Comment by ncw** [comment]
Do you want to have a go at implementing this?

If so then the first step would be to add a `Remount bool` here

https://github.com/rclone/rclone/blob/90d23139f626bcaa3a7519698773faac86cb9afb/cmd/mountlib/mount.go#L31

Then add a flag `--remount` pointing at it here

https://github.com/rclone/rclone/blob/90d23139f626bcaa3a7519698773faac86cb9afb/cmd/mountlib/mount.go#L125

The `-o remount` should get translated into a `--remount` flag.

Then somewhere just before here

https://github.com/rclone/rclone/blob/90d23139f626bcaa3a7519698773faac86cb9afb/cmd/mountlib/mount.go#L247

You check `m.MountOpt.Remount` and if it was set you'd unmount the mountpoint first. I'm not sure exactly of the best way you'd do this... You could call `fusermount -zu /mountpoint` which would work fine on linux. Perhaps we should only support this flag under linux as it is too dificult otherwise.

**Comment by Weathercold** [comment]
Sadly I can't code in Go, and I don't have time to learn it 😅

**Comment by KDreynolds** [comment]
Ill take a crack at this one.

**Comment by Bohreromir** [comment]
Hi @KDreynolds , have you had any success working on this?

**Comment by TimothyLoyer** [comment]
There had been a PR that looked like it was nearing completion, but it was closed by the author on July 1st. 😭
